tests/integration/components/dialog-test.js

```diff
@@ -1,8 +1,13 @@
 import { module, test, todo } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { click, render, setupOnerror } from '@ember/test-helpers';
-import triggerKeyEvent from '@ember/test-helpers/dom/trigger-key-event';
+import {
+  click,
+  render,
+  setupOnerror,
+  triggerKeyEvent,
+} from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
+import userEvent from '@testing-library/user-event';
```

Comment:
As `triggerEvent` and `triggerKeyEvent` don't simulate changing focus element - we need some emulator.
In `react` they created own script:
[link](https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/test-utils/interactions.ts#L291)

Reply:
I've not used `@testing-library`. This is preferable over re-implementing the same code that you linked to in the headless react lib?

Reply:
Not sure, but regarding that is only dev dependency, I use it

Reply:
This does feel a little odd to me too, though I honestly can't determine if it's actually _is_ weird or if it's just not a pattern that I've seen before. These packages are meant to be framework agnostic, so it should be fine, I think? 

Maybe it makes more sense to borrow that helper function from the React implementation and use that ourselves, and avoid the extra testing packages?

Reply:
It should work with any framework. 
About copying and pasting from react, regarding that they use `jest` there are some framework specific code:
[link](https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/test-utils/interactions.ts#L160)
so copy and paste will not work in our work.